price-feeder/config/currency_provider_tracker.go

```diff
@@ -0,0 +1,175 @@
+	coinGeckoTickersEndpoint = "tickers"
+	trackingPeriod           = time.Hour * 24
+)
+
+type (
+	// CurrencyProviderTracker queries the CoinGecko API for all the exchanges that
+	// support the currency pairs set in the price feeder config. It will poll the API
+	// every 24 hours to log any new exchanges that were added for a given currency.
+	//
+	// REF: https://www.coingecko.com/en/api/documentation
+	CurrencyProviderTracker struct {
+		logger              zerolog.Logger
+		pairs               []CurrencyPair
+		coinIDSymbolMap     map[string]string   // ex: map["ATOM"] = "cosmos"
+		CurrencyProviders   map[string][]string // map of price feeder currencies and what exchanges support them
+		CurrencyProviderMin map[string]int      // map of price feeder currencies and min required providers for them
+	}
+
+	// List of assets on CoinGecko and their corresponding id and symbol.
+	CoinGeckoCoinList struct {
+		ID     string `json:"id"`     // ex: "cosmos"
+		Symbol string `json:"symbol"` // ex: "ATOM"
+	}
+
+	// CoinGecko ticker shows market data for a given currency pair including what
+	// exchanges they're on.
+	CoinGeckoCoinTickerResponse struct {
+		Tickers []CoinGeckoCoinTicker `json:"tickers"`
+	}
+	CoinGeckoCoinTicker struct {
+		Base   string              `json:"base"`   // CurrencyPair.Base
+		Target string              `json:"target"` // CurrencyPair.Quote
+		Market CoinGeckoCoinMarket `json:"market"`
+	}
+	CoinGeckoCoinMarket struct {
+		Name string `json:"name"` // ex: Binance
+	}
+)
+
+func NewCurrencyProviderTracker(
+	ctx context.Context,
+	logger zerolog.Logger,
+	pairs ...CurrencyPair,
+) (*CurrencyProviderTracker, error) {
+	currencyProviderTracker := &CurrencyProviderTracker{
+		logger:              logger,
+		pairs:               pairs,
+		coinIDSymbolMap:     map[string]string{},
+		CurrencyProviders:   map[string][]string{},
+		CurrencyProviderMin: map[string]int{},
+	}
+
+	if err := currencyProviderTracker.setCoinIDSymbolMap(); err != nil {
+		return nil, err
+	}
+
+	if err := currencyProviderTracker.setCurrencyProviders(); err != nil {
+		return nil, err
+	}
+
+	currencyProviderTracker.setCurrencyProviderMin()
+
+	go currencyProviderTracker.trackCurrencyProviders(ctx)
+
+	return currencyProviderTracker, nil
+}
+
+func (t *CurrencyProviderTracker) getCurrencyProviders() map[string][]string {
+	return t.CurrencyProviders
+}
+
+func (t *CurrencyProviderTracker) getCurrencyProviderMin(base string) int {
+	return t.CurrencyProviderMin[base]
+}
+
+func (t *CurrencyProviderTracker) logCurrencyProviders() {
+	for currency, providers := range t.CurrencyProviders {
+		t.logger.Info().Msg(fmt.Sprintf("Providers supporting %s: %v", currency, providers))
+	}
+}
+
+// setCoinIDSymbolMap gets list of assets on coingecko to cross reference coin symbol to id.
+func (t *CurrencyProviderTracker) setCoinIDSymbolMap() error {
+	resp, err := http.Get(coinGeckoRestHost + coinGeckoRestPath + coinGeckoListEndpoint)
```

Comment:
let's use interpolation here so that we don't have to have slashes in `coinGeckoRestHost` & `coinGeckoRestPath`, and it looks a bit cleaner

Reply:
Using interpolation and put coinGeckoRestHost & coinGeckoRestPath together into coinGeckoRestUrl. Lmk how it looks!